src/clj/rems/api/applications_v2.clj

```diff
@@ -33,25 +37,32 @@
                            :get-license get-license
                            :get-user get-user})
 
+;; short-lived cache to speed up pollers which get the application
+;; repeatedly for each event instead of building their own projection
+(mount/defstate application-cache
+  :start (atom (cache/ttl-cache-factory {} :ttl 10000)))
+
 (defn get-unrestricted-application
   "Returns the full application state without any user permission
    checks and filtering of sensitive information. Don't expose via APIs."
   [application-id]
-  (let [events (applications/get-dynamic-application-events application-id)]
+  (let [events (applications/get-dynamic-application-events application-id)
```

Comment:
Getting a single application from the database is fast and doesn’t need caching (except in the pollers which naively do the get once for each event - refactoring the pollers could avoid that).

In the case of reading all applications, even then reading the events is very fast - it’s a sequential read from disk (though the indexes might not yet be optimal, so it’s maybe not sequential at the moment). What takes longest is the injections, because it does lots of small DB queries.